### ISSUE TITLE
Backslash as path separator should not be allowed in FMUs

### ISSUE THREAD

**Issue by t-sommer** [modelica/fmi-standard#547]
Using backslashes (`\`) in the paths of the contained files causes problems on Linux and macOS and should not be allowed. See also [fmi-cross-check #43](https://github.com/modelica/fmi-cross-check/issues/43).

**Comment by pmai** [comment]
It already is prohibited by the ZIP Specification, which is referenced by the standard. However given the frequency this crops up, I think an additional sentence reiterating this (as well as the common ./ prefix, which is also not sanctioned by the ZIP Specification) would be a good idea (this was one of many common issues in the FMI Implementers Best Practice Guide by ProStep which are still relevant).

One might also want to add a reference to the zip slip vulnerability https://snyk.io/research/zip-slip-vulnerability, since it is fairly easy to fall into that trap with naive use of zip libs/external programs.

**Comment by chrbertsch** [comment]
This is a 2.0.1 clarfication that should be merged in to 3.0 also, right?

**Comment by t-sommer** [comment]
The [.ZIP File Format Specification](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) is actually quite clear about this (thanks for the hint @pmai):

```
4.4.17 file name: (Variable)

       4.4.17.1 The name of the file, with optional relative path.
       The path stored MUST NOT contain a drive or
       device letter, or a leading slash.  All slashes
       MUST be forward slashes '/' as opposed to
       backwards slashes '\' for compatibility with Amiga
       and UNIX file systems etc.  If input came from standard
       input, there is no file name field. 
```

Should we close this?

**Comment by chrbertsch** [comment]
As several implementers have made this mistake, I suggest to either include a hint as non-normative text, or start an "implementers guide" or "technical faq" on the fmi website including this issue.

**Comment by andreas-junghanns** [comment]
Let's not do that, please: one document is hard enough to keep consistent; two is impossible. We have seen this with cross-check notes, notes for the standard etc. We have a good mechanism: Non-normative text. If what we have there is not perfect, let's fix this, but in the document. If we tag it well in AsciiDoc, the we can possibly turn it on/off for certain exports, but we should keep it one source with all reasonably close together. And for the specific case here, this won't be difficult.

Note: This is why I was opposed to having ProStep publish the otherwise valuable implementers guide: The place for this kind of information is inside the document and if we missed some of these additions, then we can always release a X.Y.1 to add things in.

**Comment by chrbertsch** [comment]
OK. Let's include a reference to the "ZIP File Format Specification" https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT and a hint in non-normative text to use forward slashes.

**Comment by chrbertsch** [comment]
Closing as included in merged PR #579
